### PR TITLE
[rfw] Support web (as JS)

### PR DESCRIPTION
**Pull request by Hixie**
## Pre-launch Checklist

Fixes https://github.com/flutter/flutter/issues/129843

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/rfw/lib/src/dart/binary.dart [hunk 1]
```diff
@@ -264,6 +276,10 @@ RemoteWidgetLibrary decodeLibraryBlob(Uint8List bytes) {
 // endianess used by this format
 const Endian _blobEndian = Endian.little;
 
+// whether we can use 64 bit APIs on this platform
+// (on JS, we can only use 32 bit APIs and integers only go up to ~2^53)
+const bool _kHas64Bits = 0x1000000000000000 + 1 != 0x1000000000000000; // 2^60
```
Comment (stuartmorgan) [inline packages/rfw/lib/src/dart/binary.dart:281]: Nit: Effective Dart says [not to use `k`](https://dart.dev/effective-dart/style#dont-use-prefix-letters), and Flutter style only says [to use it for global constants](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#minimize-the-visibility-scope-of-constants)

(I would strongly vote for changing the Flutter style guide to remove the `_k...` example and change "It’s not necessary to add the k prefix to non-global constants." to explicitly forbid it, to reduce pointless diff from Effective Dart.)

File: packages/rfw/test/utils.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import 'package:flutter/foundation.dart';
+
+// Detects if we're running the tests on the main channel.
+//
+// This is useful for _tests_ that depend on _Flutter_ features that have not
+// yet rolled to stable. Avoid using this to skip tests of _RFW_ features that
+// aren't compatible with stable. Those should wait until the stable release
+// channel is updated so that RFW can be compatible with it.
+bool get isMainChannel {
```
Comment (stuartmorgan) [inline packages/rfw/test/utils.dart:15]: Is the Flutter channel renaming actually happening soon? If not, this seems like confusing naming.

**Review by stuartmorgan — APPROVED** [review]
LGTM with some optional nits.

File: packages/rfw/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 1.0.12
 
+* Improved web compatibility.
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/rfw/CHANGELOG.md:3]: fixed

File: packages/rfw/lib/src/dart/binary.dart (hunk 1 above)
Comment (Hixie) [reply packages/rfw/lib/src/dart/binary.dart:281]: Fixed.

(go for it!)

**Comment by Hixie** [comment]
updated per comment, will autoland on green. Feel free to remove the `autosubmit` label if you want additional changes (e.g. if you want the main/master thing changed).

**Pull request merged**